### ISSUE TITLE
Add DSL construct to set the 'group' for all projects of a build in settings 

### ISSUE THREAD

**Issue by jjohannes** [gradle/gradle#15372]
### Why do we want that?
* Replacement for `allprojects { group = "..." }`
* If we know the group early, we do not need to configure the build to determine the group when adding substitutions for included builds. After #14562, this is the only thing stopping us from not eagerly configuring all builds in a composite.
* Usually the group is static and thus can be defined early

### How could it look

We could either add a `group.set("...")` field directly to the settings toplevel or introduce a DSL block like `identity { group.set("...") }`. 

### Notes
* Setting the `group` in a project would stay for now and using it would override the value from settings for individual projects.
* The support for automatically register substitution for groups that are set "late" will be deprecated. Users still have the option to define individual substitutions as part of an `includeBuild(...) {}` statement.
* `version` stays in build script as this more dynamic. It might require computation during the build, and/or sharing of artifacts between builds (e.g. timestamp). There is no particular reason to know the version early, as it is not needed for composite builds substitution.
*  An `identity {}` block could contain additional things in the future. For example, a `buildName` property to set a constant name for the build to identify it - currently `rootProject.name` is (miss)used for this in several places.

**Comment by melix** [comment]
IMO this shouldn't be limited to the `group`. There should be a naming strategy, group and artifact, in the settings probably. Something that we could derive from the project path too.

**Comment by bamboo** [comment]
Take into account that some use cases might require wiring in `group` (and `version`) lazily (https://github.com/gradle/gradle/issues/15088).

**Comment by melix** [comment]
Still relevant
